src/lib/Iqons.js

```diff
@@ -0,0 +1,223 @@
+class Iqons {
+
+    /* Public API */
+
+    /**
+     * @param {string} text
+     */
+    static async svg(text) {
+        const hash = this._hash(text);
+        return this._svgTemplate(
+            parseInt(hash[0]),
+            parseInt(hash[2]),
+            parseInt(hash[3] + hash[4]),
+            parseInt(hash[5] + hash[6]),
+            parseInt(hash[7] + hash[8]),
+            parseInt(hash[9] + hash[10]),
+            parseInt(hash[11])
+        );
+    }
+
+    /**
+     * @param {string} text
+     */
+    static async toDataUrl(text) {
+        const base64string = btoa(await this.svg(text));
+        return `data:image/svg+xml;base64,${base64string.replace(/#/g, '%23')}`;
+    }
+
+    /**
+     * @param {string} [color]
+     * @param {number} [strokeWidth]
+     */
+    static placeholder(color, strokeWidth) {
+        color = color || '#bbb';
+        strokeWidth = strokeWidth || 1;
+        return `<svg viewBox="0 0 160 160" width="160" height="160" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/2000/xlink" >
+    <path fill="none" stroke="${color}" stroke-width="${2 * strokeWidth}" transform="translate(0, 8) scale(0.5)" d="M251.6 17.34l63.53 110.03c5.72 9.9 5.72 22.1 0 32L251.6 269.4c-5.7 9.9-16.27 16-27.7 16H96.83c-11.43 0-22-6.1-27.7-16L5.6 159.37c-5.7-9.9-5.7-22.1 0-32L69.14 17.34c5.72-9.9 16.28-16 27.7-16H223.9c11.43 0 22 6.1 27.7 16z"/>
+    <g transform="scale(0.9) translate(9, 8)">
+        <circle cx="80" cy="80" r="40" fill="none" stroke="${color}" stroke-width="${strokeWidth}" opacity=".9"></circle>
+        <g opacity=".1" fill="#010101"><path d="M119.21,80a39.46,39.46,0,0,1-67.13,28.13c10.36,2.33,36,3,49.82-14.28,10.39-12.47,8.31-33.23,4.16-43.26A39.35,39.35,0,0,1,119.21,80Z"/></g>\`
+    </g>
+</svg>`;
+    }
+
+    /**
```

Comment:
can be removed in keyguard

Reply:
Placeholders are set as the image now as long as no address is set.